### PR TITLE
Fix DataPack Overwrite Regression

### PR DESCRIPTION
**Pull request by Headline**
Some tests passed with the implementation prior to this commit, but those were edge cases. Good older behavior is now fully restored

### CODE REVIEW

File: core/logic/CDataPack.cpp [hunk 1]
```diff
@@ -219,12 +219,16 @@ bool CDataPack::RemoveItem(size_t pos)
 	{
 		pos = position;
 	}
-
 	if (pos >= elements.length())
 	{
 		return false;
 	}
 
+	if (pos < position) // we're deleting under us, step back
```
Comment (asherkin) [inline core/logic/CDataPack.cpp]: Should this not be `<=`? My brain has trouble following the logic here so I can't be sure.

Comment (Headline) [reply core/logic/CDataPack.cpp]: if `pos = position`, we're still deleting in front of us. (`position` is the index to the *next* read/write index in the array.

File: core/logic/CDataPack.cpp [hunk 2]
```diff
@@ -219,12 +219,16 @@ bool CDataPack::RemoveItem(size_t pos)
...
 	}
-
 	if (pos >= elements.length())
 	{
 		return false;
 	}
 
+	if (pos < position) // we're deleting under us, step back
+	{
+		position--;
```
Comment (KyleSanderson) [inline core/logic/CDataPack.cpp]: pre to post?

File: core/logic/CDataPack.cpp (hunk 1 above)
Comment (KyleSanderson) [inline core/logic/CDataPack.cpp]: why not just set this to set position to pos?

Comment (Headline) [reply core/logic/CDataPack.cpp]: if we’re deleting far behind us, we shouldn’t  change our pack position. Only scoot us down to account for it.

if cell func string was written, and we deleted cell, our position shouldn’t reset to be 0 and we should still be after the func write

File: core/logic/CDataPack.cpp (hunk 2 above)
Comment (Headline) [reply core/logic/CDataPack.cpp]: no comprendo
